src/main/java/io/javalin/core/CachedRequestWrapper.kt

```diff
@@ -23,12 +23,16 @@ class CachedRequestWrapper(request: HttpServletRequest, private val maxCacheSize
     // Do not read unless we have to
     private val cachedBytes: ByteArray by lazy { super.getInputStream().readBytes() }
 
-    override fun getInputStream(): ServletInputStream =
-            if (chunkedTransferEncoding || maxCacheSize < size) {
-                super.getInputStream()
-            } else {
-                CachedServletInputStream(cachedBytes)
-            }
+    override fun getInputStream(): ServletInputStream {
+        if (exceptionOnRequestCacheIsSmall && maxCacheSize < size)
+            throw Exception("Request cache size is too small for request body.")
```

Comment:
Throwing Exception directly feels wrong for me personally. Is there a better Exception available?

Reply:
fixed in 1fa7d2106cf60572c86edcef229c77e667e561df